tests/components/advantage_air/__init__.py

```diff
@@ -54,6 +59,27 @@ async def add_mock_config(hass: HomeAssistant) -> MockConfigEntry:
     )
 
     entry.add_to_hass(hass)
-    await hass.config_entries.async_setup(entry.entry_id)
+    if platforms is None:
+        await hass.config_entries.async_setup(entry.entry_id)
+    else:
+        with patch("homeassistant.components.advantage_air.PLATFORMS", platforms):
+            await hass.config_entries.async_setup(entry.entry_id)
     await hass.async_block_till_done()
     return entry
+
+
+def assert_entities(
+    hass: HomeAssistant,
+    entry_id: str,
+    entity_registry: er.EntityRegistry,
+    snapshot: SnapshotAssertion,
+) -> None:
+    """Test that all entities match their snapshot."""
+
+    entity_entries = er.async_entries_for_config_entry(entity_registry, entry_id)
+
+    assert entity_entries
+    for entity_entry in entity_entries:
+        assert entity_entry == snapshot(name=f"{entity_entry.entity_id}-entry")
+        assert (state := hass.states.get(entity_entry.entity_id))
+        assert state == snapshot(name=f"{entity_entry.entity_id}-state")
```

Comment:
Why don't you use snapshot_entities?

Reply:
This was the first integration I ever wrote, before I knew about snapshots. I definetly could rework the other platforms to use snapshots too, but was trying to have this PR address the test flakyness in binary sensor and sensor specifically.